files/scripts/syncd.sh

```diff
@@ -155,6 +174,12 @@ stop() {
         debug "Stopped pmon service"
     fi
 
+    if [[ x$sonic_asic_platform == x"mellanox" ]]; then
+        # In case of SAI failure the last line of sairedis.rec would contain switch_shutdown_request
+        if tail -1 /var/log/swss/sairedis.rec | grep -q switch_shutdown_request; then
```

Comment:
only the last line? should be a few more lines?

Reply:
@lguohan  In case of syncd triggered shutdown it will always be the last lag before reboot Here are examples

2022-07-27.16:50:06.872479|n|switch_shutdown_request|{"switch_id":"oid:0x0"}|
2022-07-27.16:51:25.278359|#|recording on: /var/log/swss/sairedis.rec
2022-07-27.16:51:25.278460|#|logrotate on: /var/log/swss/sairedis.rec
--
2022-07-27.16:52:36.845899|n|switch_shutdown_request|{"switch_id":"oid:0x0"}|
2022-07-27.16:54:00.166846|#|recording on: /var/log/swss/sairedis.rec
2022-07-27.16:54:00.166957|#|logrotate on: /var/log/swss/sairedis.rec
--
2022-07-27.17:17:50.873397|n|switch_shutdown_request|{"switch_id":"oid:0x0"}|
2022-07-27.17:19:08.236351|#|recording on: /var/log/swss/sairedis.rec
2022-07-27.17:19:08.236546|#|logrotate on: /var/log/swss/sairedis.rec
--
2022-07-27.17:30:37.724769|n|switch_shutdown_request|{"switch_id":"oid:0x0"}|
2022-07-27.17:31:55.674532|#|recording on: /var/log/swss/sairedis.rec
2022-07-27.17:31:55.674653|#|logrotate on: /var/log/swss/sairedis.rec
--
2022-07-27.17:34:50.444261|n|switch_shutdown_request|{"switch_id":"oid:0x0"}|
2022-07-27.17:36:07.846780|#|recording on: /var/log/swss/sairedis.rec
2022-07-27.17:36:07.846886|#|logrotate on: /var/log/swss/sairedis.rec